app/services/homepage/fetch_tag_flares.rb

```diff
@@ -0,0 +1,35 @@
+module Homepage
+  class FetchTagFlares
+    ATTRIBUTES = %i[name bg_color_hex text_color_hex].freeze
+
+    def self.call(articles)
+      articles_tag_flares = articles_tag_flares(articles)
+
+      tags = Tag.where(name: articles_tag_flares.keys).select(*ATTRIBUTES).index_by(&:name)
+
+      tag_flares = {}
+      articles_tag_flares.each do |tag_name, article_ids|
+        article_ids.each do |article_id|
+          tag_flares[article_id] = tags[tag_name].as_json(only: ATTRIBUTES)
+        end
+      end
+
+      tag_flares
+    end
+
+    def self.articles_tag_flares(articles)
+      flare_tags_for_articles = Hash.new { |hash, key| hash[key] = [] }
+
+      articles.pluck(:id, :cached_tag_list).map do |article_id, cached_tag_list|
+        tags = cached_tag_list.split(", ")
+        flare_tag = ::Constants::Tags::FLARE_TAG_NAMES.to_set.intersection(tags).first
```

Comment:
I'm not sure how many articles you have but turning the same constant repeatedly into a set seems a bit inefficient. Maybe move the set creation before the `map`, e.g.

```ruby
flare_tag_names = ::Constants::Tags::FLARE_TAG_NAMES.to_set
articles.pluck(:id, :cached_tag_list).map do |article_id, cached_tag_list|
  tags = cached_tag_list.split(", ")
  flare_tag = flare_tag_names.intersection(tags).first
```